homeassistant/components/siren/__init__.py

```diff
@@ -0,0 +1,135 @@
+"""Component to interface with various sirens/chimes."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any, TypedDict, cast, final
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import SERVICE_TOGGLE, SERVICE_TURN_OFF, SERVICE_TURN_ON
+from homeassistant.core import ServiceCall
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.config_validation import (  # noqa: F401
+    PLATFORM_SCHEMA,
+    PLATFORM_SCHEMA_BASE,
+)
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from .const import (
+    ATTR_AVAILABLE_TONES,
+    ATTR_DURATION,
+    ATTR_TONE,
+    ATTR_VOLUME_LEVEL,
+    DOMAIN,
+    SUPPORT_DURATION,
+    SUPPORT_TONES,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_SET,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+TURN_ON_SCHEMA = {
+    vol.Optional(ATTR_TONE): cv.string,
+    vol.Optional(ATTR_DURATION): cv.positive_float,
+    vol.Optional(ATTR_VOLUME_LEVEL): cv.small_float,
+}
+
+
+class SirenTurnOnServiceParametersDataType(TypedDict, total=False):
```

Comment:
I don't think we should `Type` or `DataType` everything, it is already typed, as it is a TypedDict.

```suggestion
class SirenTurnOnServiceParameters(TypedDict, total=False):
```

Clean and descriptive.